pandas/core/indexes/datetimes.py

```diff
@@ -946,7 +945,7 @@ def insert(self, loc, item):
                     freq = self.freq
                 elif (loc == len(self)) and item - self.freq == self[-1]:
                     freq = self.freq
-            item = _to_M8(item, tz=self.tz)
+            item = item.asm8
```

Comment:
can we remove _to_M8?

Reply:
> can we remove _to_M8?

IIRC there's still one more usage to get rid of after this.  I'm eager to get rid of it, as im pretty sure it is hiding a bunch of corner case bugs.